### PR TITLE
Dev share ui

### PR DESCRIPTION
**Pull request by EnBakir**
Implemented the share toolbox menu.

### CODE REVIEW

File: src/modules/menu/components/toolBar/ToolBar.js [hunk 1]
```diff
@@ -97,6 +98,11 @@ export class ToolBar extends BaElement {
 			toggleTool(toolId);
 		};
 
+		const toggleShareTool = () => {
+			const toolId = ShareToolContent.tag;
+			toggleTool(toolId);
+		};
+
 		const translate = (key) => this._translationService.translate(key);
```
Comment (taulinger) [inline src/modules/menu/components/toolBar/ToolBar.js:106]: Don't forget to add the test in 
`it('toggles and switches the tools'`

File: src/modules/toolbox/components/shareToolContent/ShareToolContent.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		this.render();
+	}
+
+	createView() {
+		const translate = (key) => this._translationService.translate(key); 
+
+		const onChange = (event) => {
+			const checked = event.target.checked;
+			if (!checked) {
+				this._root.querySelector('.preview_button').classList.add('disabled-preview');
```
Comment (taulinger) [inline src/modules/toolbox/components/shareToolContent/ShareToolContent.js:68]: No test for this path

**Review by taulinger — COMMENTED** [review]
Nice proceedings!

@alsturm Does it make sense to add and include a common CSS for all ToolContainer?

**Review by taulinger — CHANGES_REQUESTED** [review]
The generated URL is not always in sync with the current state.
E.g. the user changes the position of the map
while the  ShareToolContent is active.

**Review by thiloSchlemmer — APPROVED** [review]

**Review by thiloSchlemmer — APPROVED** [review]

**Review by taulinger — APPROVED** [review]

**Comment by taulinger** [comment]
We should convert it to a PR not until @alsturm has optimized the layout.

**Review by thiloSchlemmer — APPROVED** [review]

File: src/modules/toolbox/components/shareToolContent/assets/envelope-solid.svg [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-envelope" viewBox="0 0 16 16">
```
Comment (taulinger) [inline src/modules/toolbox/components/shareToolContent/assets/envelope-solid.svg:1]: @alsturm License is missing

**Review by taulinger — DISMISSED** [review]
Almost done, see inline comments ...

**Review by alsturm — DISMISSED** [review]

**Review by thiloSchlemmer — APPROVED** [review]

**Review by taulinger — COMMENTED** [review]
We now have a different approach of when to show which kind of share possibility.
So let's start a round of reviews...

**Review by thiloSchlemmer — APPROVED** [review]

**Pull request merged**